### PR TITLE
Import build uki command from enki

### PR DESCRIPTION
**Pull request by jimmykarily**
This is just a copy-over from the enki code. Some minor changes were made so that no function expects a global viper object. They now take everything they need as function parameters.

Steps to deprecate the enki command:

- [x] Port the code to Auroraboot and make it work the same
- [x] Migrate whatever tests can be migrated
- [ ] Merge
- [ ] Refactor the code to make it cleaner and re-usable
- [ ] Move any leftovers from enki to Auroraboot (because this PR still uses things from enki, like constants and functions).

When the rest of the enki commands have been migrated (`genkey` and `sysext` remaining), we can archive the enki repo to make sure no fixes are made there instead of Auroraboot.

### CODE REVIEW

**Comment by jimmykarily** [comment]
I printed the files as they are being copied with `mcopy` :

```
2024-11-07T09:29:12Z INF Copying files in the img file
************************* f = /tmp/enki-build-uki-3252635559/BOOTX64.EFI
************************* f = /tmp/enki-build-uki-3252635559/norole.efi
************************* f = /tmp/enki-build-uki-3252635559/My_Entry.efi
************************* f = /tmp/enki-build-uki-3252635559/My_Other_Entry.efi
************************* f = /tmp/enki-build-uki-3252635559/Something_Here_Is_Not_Right.efi
************************* f = /tmp/enki-build-uki-3252635559/loader.conf
************************* f = /tmp/enki-build-uki-3252635559/norole.conf
************************* f = /tmp/enki-build-uki-3252635559/My_Entry.conf
************************* f = /tmp/enki-build-uki-3252635559/My_Other_Entry.conf
************************* f = /tmp/enki-build-uki-3252635559/Something_Here_Is_Not_Right.conf
************************* f = /home/dimitris/workspace/kairos/AuroraBoot/e2e/assets/keys/PK.der
************************* f = /home/dimitris/workspace/kairos/AuroraBoot/e2e/assets/keys/KEK.der
************************* f = /home/dimitris/workspace/kairos/AuroraBoot/e2e/assets/keys/db.der
************************* f = /home/dimitris/workspace/kairos/AuroraBoot/e2e/assets/keys/PK.auth
************************* f = /home/dimitris/workspace/kairos/AuroraBoot/e2e/assets/keys/KEK.auth
************************* f = /home/dimitris/workspace/kairos/AuroraBoot/e2e/assets/keys/db.auth
```
The file name looks correct until the last minute. I wonder if it's the ISO filesystem somehow lying to me (not having filename case information on the the files or something).

**Comment by jimmykarily** [comment]
Never mind, we decided to go lowercase with all files. Fat can't be trusted to maintain the case.

File: e2e/genkey_test.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
+package e2e_test
```
Comment (jimmykarily) [inline e2e/genkey_test.go:1]: Not being run yet by the pipeline. Will enable it when we port this command.

Comment (jimmykarily) [reply e2e/genkey_test.go]: removed

**Review by Itxaka — APPROVED** [review]
looks good. I guess the refactor of things wil come later? Seeing some things in the build-uki iso that may be better in a separate utils package or something similar as having a command being 1000 lines its a bit terrible xD

**Comment by jimmykarily** [comment]
> looks good. I guess the refactor of things wil come later? Seeing some things in the build-uki iso that may be better in a separate utils package or something similar as having a command being 1000 lines its a bit terrible xD

Yup, it's as terrible as it gets :D I didn't want to do the refactoring together with the porting because it would make the review of the refactoring part impossible. Let's first migrate the commands.

**Pull request merged**